lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -29,7 +29,21 @@ export class PublicClientApplication extends ClientApplication implements IPubli
 
     /**
      * @constructor
-     * Constructor for the PublicClientApplication used to instantiate the PublicClientApplication object
+     * Constructor for the PublicClientApplication used to instantiate the PublicClientApplication object. To be used
+     * internally within the class or any class that extends it but not externally
+     *
+     * @param configuration object for the MSAL PublicClientApplication instance
+     */
+    protected constructor(configuration: Configuration) {
```

Comment:
breaking change?

Reply:
Yes, it's a breaking change. Why do we want to avoid breaking changes in major releases?

Reply:
@konstantin-msft we decided not to ship any API surface changes for v3 to make migration easier. Only breaking changes we're shipping are behavior changes.

Reply:
I wasn't aware of that. Thanks @hectormmg .

Reply:
I think this one needs to be in for the JS/WAM work.  It's the only API breaking change I'm willing to take and throwing if we're not initialized (suggestion I had above) should help folks detect that this change is needed when they run their build tests vs when they are in prod.

Reply:
@peterzenz agreed, it's actually not something we can circumvent with the JS/WAM as default change. It will also mostly be a single-place update for most developers, so the surface change delta is minimal anyway.

Reply:
We can keep the API surface the same but also start requiring people to call initialize (by throwing if it hasn't been called). I do like the changes proposed in this PR but if the desire is to make migration as easy as possible we can table the static `getInstance` pattern until v4. 

I don't really feel strongly about surface changes in v3 so either approach is good for me, just wanted to throw that option out there for those who do

Reply:
From my perspective requiring people to call `initialize()` is still a breaking change. I would really prefer to use a static method. It seems like an easy migration to me because it's a one-line code change for customers.

Reply:
Agreed it's still breaking. The difference is: people who are calling initialize today don't have to make any changes at all.

Because the static method turns PCA construction from sync to async it could be much more than a one-line change for people depending on how their apps are designed today. Don't get me wrong, I like the approach but the more I think about it the more I think it may be better suited for v4 given the desire to keep v3 light on changes.

Reply:
Well, in that case I can leave `createPublicClientApplication()` static method as an alternative approach to get initialized app instance.

Reply:
Sure that's a good idea, I'm ok with that.

Reply:
Wrapping up:
1. Public client `initialize()` remains public. We mark it as mandatory in v3.
2. New method `createPublicClientApplication()` is added as an alternative to get initialized app client.
3. Condition checks to validate that app is initialized are already in place, so nothing needs to be changed here.

@tnorling @peterzenz @hectormmg please let me know if you are OK with that.

Reply:
I'm ok with that, however, minor clarification regarding point 3. I believe the checks are only done today when allowBrokering is true. We'll want to make sure this is always checked regardless of config.

Reply:
Late to the party but adding the comments. Can we not "initialize" if "allow brokering" is true? I do not like the idea of two ways to initialize the PCA. I would rather leverage what @shoatman did in his PR and have this as an "operating context" based on "allow brokering" set up.